dev-cli/container/src/core/main.c

```diff
@@ -12,11 +15,148 @@
 #include "lauxlib.h"
 #endif
 
+#include "gmp.h"
+
 #ifdef __cplusplus
 extern "C"
 {
 #endif
 
+bool compute_mod_exp(const char *base_input, const char *exp_input, const char *mod_input, const char *expected_input) {
+    mpz_t base, exp, mod, large_exp, result, expected;
+    bool match = false;
+
+    // Initialize big integers
+    mpz_init(base);
```

Comment:
Just threw these right into the `main.c` file for getting stuff working quickly. Can reorg as needed.